### PR TITLE
EIP-3860: Clarify rules inclusion and failure rules

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

**Comment by axic** [comment]
@jochem-brouwer this should have fixed the concerns you had.

**Comment by jochem-brouwer** [comment]
Great, thanks a lot @axic 😄 

I think my main confusion comes from the word "invalid". We also have `INVALID` opcode which consumes all gas. So for me, an "invalid transaction" either means (1) invalid; cannot be put in block  because then block is invalid (cannot be included in the blockchain) or (2) it immediately consumes all gas upon execution. But the extra details make it very clear, thanks 👍

File: EIPS/eip-3860.md [hunk 1]
```diff
@@ -52,11 +52,10 @@ We define `initcode_cost(initcode)` to equal `INITCODE_WORD_COST * ceil(len(init
...
-2. For a create transaction, extend the transaction data cost formula to include `initcode_cost(initcode)`.
-3. If length of `initcode` to `CREATE` or `CREATE2` instructions exceeds `MAX_INITCODE_SIZE`, execution results in an exceptional abort.
-4. For the `CREATE` and `CREATE2` instructions charge an extra gas cost equaling to `initcode_cost(initcode)`. This cost is deducted before the calculation of the resulting contract address and the execution of `initcode`. (Note that this means before or at the same time as the hashing cost is applied in `CREATE2`.)
-
+1. If length of transaction data (`initcode`) in a create transaction exceeds `MAX_INITCODE_SIZE`, transaction is invalid. (*Note that this is similar to transactions considered invalid for not meeting the intrinsic gas cost requirement.*)
+2. For a create transaction, extend the transaction data cost formula to include `initcode_cost(initcode)`. Note that this is included in transaction intrinsic cost, i.e. transaction with not enough gas to cover initcode cost is invalid.
+3. If length of `initcode` to `CREATE` or `CREATE2` instructions exceeds `MAX_INITCODE_SIZE`, instruction results in `0` value pushed on stack.
```
Comment (chfast) [inline EIPS/eip-3860.md:57]: I wander if we should additionally mention the "instruction execution terminates with result `0` pushed on stack".

File: EIPS/eip-3860.md [hunk 2]
```diff
@@ -52,11 +52,10 @@ We define `initcode_cost(initcode)` to equal `INITCODE_WORD_COST * ceil(len(init
...
+4. For the `CREATE` and `CREATE2` instructions charge an extra gas cost equaling to `initcode_cost(initcode)`. This cost is deducted before the calculation of the resulting contract address and the execution of `initcode`. (*Note that this means before or at the same time as the hashing cost is applied in `CREATE2`.*)
```
Comment (chfast) [inline EIPS/eip-3860.md:58]: Points 1, 2 and 4 contains _Notes_, but all notes are done in different style.

Comment (axic) [reply EIPS/eip-3860.md:58]: Noted 😅

**Pull request merged**